### PR TITLE
[daggy-u] update usage of `context.add_output_metadata` with `MaterializeResult`

### PR DESCRIPTION
**Pull request by cmpadden**
## Summary & Motivation

With Dagster University being upgraded to v1.6, we should update our usage of `context.add_output_metadata` to returning a `MaterializeResult` with the `metadata` defined as a parameter.

The corresponding changes to `project-dagster-university` can be found here:

https://github.com/dagster-io/project-dagster-university/pull/8

### CODE REVIEW

**Comment by cmpadden** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/19575?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

Current dependencies on/for this PR:
* `master`
  * **PR #19566** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/19566?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #19575** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/19575?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/19575?utm_source=stack-comment).

File: docs/dagster-university/pages/dagster-essentials/extra-credit/asset-metadata-as-markdown.md [hunk 1]
```diff
@@ -126,13 +126,13 @@ In Lesson 9, you created the `adhoc_request` asset. During materialization, the
    2. `base64.b64encode` encodes the image’s binary data (`image_data`) into base64 format.
    3. Next, the encoded image data is converted to a UTF-8 encoded string using the `decode` function.
    4. Next, a variable named `md_content` is created. The value of this variable is a Markdown-formatted string containing a JPEG image, where the base64 representation of the image is inserted.
-   5. Using `context.add_output_metadata`, the image is passed in as metadata. The metadata will have a `preview` label in the Dagster UI.
+   5. We are able to include the metadata on the asset by returning a `MaterializeResult` instance with the image passed in as metadata. The metadata will have a `preview` label in the Dagster UI.
```
Comment (erinkcochran87) [inline docs/dagster-university/pages/dagster-essentials/extra-credit/asset-metadata-as-markdown.md:129]:
```suggestion
   5. To include the metadata on the asset, we returned a `MaterializeResult` instance with the image passed in as metadata. The metadata will have a `preview` label in the Dagster UI.
```

Kinda nitpicky (sorry!) but this small change makes the copy fit more with how the other bullets are currently structured.

File: docs/dagster-university/pages/dagster-essentials/extra-credit/coding-practice-metadata-taxi-zones-file.md [hunk 2]
```diff
@@ -18,7 +18,7 @@ The metadata you built should look similar to the code contained in the **View a
 @asset(
     group_name="raw_files",
 )
-def taxi_zones_file(context):
+def taxi_zones_file():
```
Comment (erinkcochran87) [inline docs/dagster-university/pages/dagster-essentials/extra-credit/coding-practice-metadata-taxi-zones-file.md:21]: D'you think we need to add or mention adding the `MaterializeResult` import? It looks like we add it earlier in the lesson to update the `taxi_trips_file` asset, but curious if you think it's worth it to do.

File: docs/dagster-university/pages/dagster-essentials/extra-credit/materialization-metadata.md [hunk 3]
```diff
@@ -14,8 +14,8 @@ Now that we’ve covered definition metadata, let’s dive into the other type o
 
 To add metadata to an asset, you need to do two things:
 
-- Use the `context.add_output_metadata` function to pass in the data
 - Use the `MetadataValue` utility class to wrap the data, ensuring it displays correctly in the UI
+- Return a `MaterializeResult` instance with the `metadata` parameter from your asset
```
Comment (erinkcochran87) [inline docs/dagster-university/pages/dagster-essentials/extra-credit/materialization-metadata.md:18]: Nit - this comes first in code, so lets have the order of this list match that.

Comment (erinkcochran87) [reply docs/dagster-university/pages/dagster-essentials/extra-credit/materialization-metadata.md:18]: Also, is it `instance` or `object`?

Comment (cmpadden) [reply docs/dagster-university/pages/dagster-essentials/extra-credit/materialization-metadata.md]: From what I'm seeing online, `instance` and `object` are mostly interchangeable. I've changed it to `object` to be aligned with the rest of the documentation.

**Comment by cmpadden** [comment]
@erinkcochran87 thank you for the feedback - I've made changes to address your comments. Let me know if there is anything else!

**Comment by cmpadden** [comment]
@erinkcochran87, have you experienced failed Vercel aliases like this in the past? My theory is that it has to do with the branch name generated by Graphite, but I'm not 100% certain.

```
Assigning alias 02-02--daggy-u-update-usage-of-context-add-output-metadata-with-MaterializeResult.dagster.dagster-docs.io to deployment dagster-docs-uk8dc91qa-elementl.vercel.app
Creating alias
Error! Invalid alias. Please confirm that the alias you provided is a valid hostname. Note: For `vercel.app`, only sub and sub-sub domains are supported.```

**Comment by erinkcochran87** [comment]
@cmpadden I have no idea - I don't use Graphite 😬

**Review by erinkcochran87 — APPROVED** [review]

**Pull request merged**